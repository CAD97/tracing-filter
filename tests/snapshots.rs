use {
    std::{fmt, fs, path::Path},
    tracing_filter::simple::Filter,
};

struct DisplayDiagnostic<'a>(&'a dyn miette::Diagnostic);

impl fmt::Display for DisplayDiagnostic<'_> {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        miette::GraphicalReportHandler::new_themed(miette::GraphicalTheme::unicode_nocolor())
            .with_links(false)
            .render_report(f, self.0)
            .or_else(|_| write!(f, "failed to pretty-print {:#?}", self.0))
    }
}

#[test]
fn snapshot_filter_parser() {
    fn callback(path: &Path) {
        let src = fs::read_to_string(path).unwrap();
        let (filter, report) = Filter::parse(&src);

        if let Some(filter) = filter {
            insta::assert_snapshot!(Some("filter"), format!("{filter:#?}"), &src);
        } else {
            insta::assert_snapshot!(Some("filter"), "(compilation failed)", &src);
        }
        if let Some(report) = report {
            let report = DisplayDiagnostic(&*report);
            insta::assert_snapshot!(Some("report"), format!("{report}"), &src);
        } else {
            insta::assert_snapshot!(Some("report"), "(no warnings)", &src);
        }
    }

    insta::glob!("test_cases/*.env_filter", callback);
}
