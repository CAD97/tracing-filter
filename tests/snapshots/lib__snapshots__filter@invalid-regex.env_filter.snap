---
source: tests/snapshots.rs
assertion_line: 23
expression: "crate::target=warn/[[]\n"
---
Filter {
    directives: [
        Directive {
            target: Some(
                "crate::target",
            ),
            level: LevelFilter::WARN,
        },
    ],
    regex: None,
}
